### ISSUE TITLE
[New rule] `unnessecary-late-keyword`

### ISSUE THREAD

**Issue by DetachHead** [dart-code-checker/dart-code-metrics#1179]
### Rule details

report an error if a `late` variable is initialized on the same line, as the `late` keyword is not required.

### What type of rule is this?

Warns about a potential problem

### Example code

```dart
late bool foo = false; // error
late bool bar; // no error
```

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### Additional comments

i raised this before #1113 but i worded it terribly. sorry!

**Comment by incendial** [comment]
@DetachHead the thing with late is that it produces a lazy computed variable and the value will be calculated only on the first usage. I'm not sure that we're able to distinguish when late is used on purpose or not. Maybe you can add more examples?

**Comment by incendial** [comment]
So I can understand when this functionality is part of `avoid-late-keword`, but as a separate rule - it will create false-positives

**Comment by DetachHead** [comment]
> the thing with late is that it produces a lazy computed variable and the value will be calculated only on the first usage. I'm not sure that we're able to distinguish when late is used on purpose or not. Maybe you can add more examples?

ah, I didn't know it worked like that. i'll close this issue then. sorry again for wasting your time with #1113

**Comment by incendial** [comment]
No problem, #1113 looks fine
